### PR TITLE
add method to conditional check for mkldnn version & use conv_auto

### PR DESCRIPTION
**Pull request by pruthviIntel**
image

```
mkldnn_verbose,info,Intel(R) MKL-DNN v0.18.0 (Git Hash 863ff6e7042cec7d2e29897fe9f0872e0888b0fc),Intel(R) Advanced Vector Extensions 512 (Intel(R) AVX-512) with AVX512BW, AVX512VL, and AVX512DQ extensions
mkldnn_verbose,create,reorder,simple:any,undef,in:f32_nchw out:f32_OIhw16i16o,num:1,64x3x3x3,0.0129395
mkldnn_verbose,exec,reorder,simple:any,undef,in:f32_nchw out:f32_OIhw16i16o,num:1,64x3x3x3,0.414062
mkldnn_verbose,create,reorder,simple:any,undef,in:f32_nchw out:f32_nChw16c,num:1,64x3x224x224,0.0119629
mkldnn_verbose,exec,reorder,simple:any,undef,in:f32_nchw out:f32_nChw16c,num:1,64x3x224x224,19.302
mkldnn_verbose,create,convolution,jit_wino_4x3:avx512_core,forward_training,fsrc:nChw16c fwei:OIhw16i16o fbia:undef fdst:nChw16c,alg:convolution_winograd,mb64_ic3oc64_ih224oh224kh3sh1dh0ph1_iw224ow224kw3sw1dw0pw1,1.84106
mkldnn_verbose,exec,convolution,jit_wino_4x3:avx512_core,forward_training,fsrc:nChw16c fwei:OIhw16i16o fbia:undef fdst:nChw16c,alg:convolution_winograd,mb64_ic3oc64_ih224oh224kh3sh1dh0ph1_iw224ow224kw3sw1dw0pw1,46.6631
mkldnn_verbose,create,reorder,jit:uni,undef,in:f32_nChw16c out:f32_nchw,num:1,64x64x224x224,0.279053
mkldnn_verbose,exec,reorder,jit:uni,undef,in:f32_nChw16c out:f32_nchw,num:1,64x64x224x224,100.219
```

### CODE REVIEW

File: src/ngraph/runtime/cpu/mkldnn_utils.cpp [hunk 1]
```diff
@@ -706,7 +706,22 @@ bool runtime::cpu::mkldnn_utils::can_use_mkldnn_batchnorm_fprop(const ngraph::No
         return false;
     }
 }
+bool runtime::cpu::mkldnn_utils::can_use_conv_auto()
```
Comment (jbobba) [inline src/ngraph/runtime/cpu/mkldnn_utils.cpp]: split this into two methods. get_mkldnn_version() and can_use_conv_auto()

Comment (pruthviIntel) [reply src/ngraph/runtime/cpu/mkldnn_utils.cpp]: sure Will do that.

**Comment by pruthviIntel** [comment]
@jbobba addressed the PR comments, can you take a look now?

**Comment by pruthviIntel** [comment]
@nbpatel can u please review it ? and approve if it looks good.

File: src/ngraph/runtime/cpu/mkldnn_utils.cpp [hunk 2]
```diff
@@ -707,6 +707,27 @@ bool runtime::cpu::mkldnn_utils::can_use_mkldnn_batchnorm_fprop(const ngraph::No
     }
 }
 
+const mkldnn_version_t* runtime::cpu::mkldnn_utils::get_mkldnn_version()
+{
+    return mkldnn_version();
+}
+
+mkldnn::algorithm runtime::cpu::mkldnn_utils::can_use_conv_auto()
```
Comment (jbobba) [inline src/ngraph/runtime/cpu/mkldnn_utils.cpp]: rename to get_conv_algo()

File: src/ngraph/runtime/cpu/mkldnn_utils.cpp [hunk 3]
```diff
@@ -707,6 +707,27 @@ bool runtime::cpu::mkldnn_utils::can_use_mkldnn_batchnorm_fprop(const ngraph::No
     }
 }
 
+const mkldnn_version_t* runtime::cpu::mkldnn_utils::get_mkldnn_version()
```
Comment (jbobba) [inline src/ngraph/runtime/cpu/mkldnn_utils.cpp]: this probably needs to be guarded by #if defined(MKLDNN_VERSION_MAJOR)...

File: src/ngraph/runtime/cpu/mkldnn_emitter.cpp [hunk 4]
```diff
@@ -243,10 +243,11 @@ mkldnn::memory::format MKLDNNEmitter::query_convolution_forward_weight_format(
     mkldnn::memory::dims mkldnn_padding_below(padding_below.begin(), padding_below.end());
     mkldnn::memory::dims mkldnn_padding_above(padding_above.begin(), padding_above.end());
 
+    mkldnn::algorithm convolution_algo = mkldnn_utils::get_conv_algo();
```
Comment (nmostafa) [inline src/ngraph/runtime/cpu/mkldnn_emitter.cpp]: We call this repeatedly for each emitter function call. Would be nice to have a static variable instead and set it once upon emitter start.

**Pull request merged**